PostHog/posthog
# chore(materialized-columns): remove dollar signs from columns

## Problem

Removing `$` from all columns is needed to upgrade CH.

Replaces: https://github.com/PostHog/posthog/pull/8855

I started by trying to revive that PR, but because of merge issues etc, it was easier to start fresh and move things over from there.

Copying context from #8855:

> Follow-up to https://github.com/PostHog/posthog/pull/8846
> Our materialization logic currently creates columns with a $ sign. According to the linked PR, clickhouse added heredoc support, breaking this feature.
> This PR introduces an alternative solution by removing $ from the column names. I'd love to roll back the linked PR and use this instead as I think this has less of a chance of becoming a frequent footgun during development.

## Changes

* Removes the `$` from the `$group_` columns + `$window_id` + `$session_id`
* Migration to create aliases for existing materialized columns that have `$` in them
* Cleans up `extra_properties` stuff in insight queries (tied in with the `$window_id` + `$session_id` changes)

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Manually tested by:
* Upgrading an old instance with `$` in columns and ensuring
    * Columns are renamed correctly (on events + persons)
    * New columns can be materialized (on events + persons)
* Setting up a new instance and ensuring:
    * Columns are renamed correctly (on events + persons)
    * New columns can be materialized (on events + persons)
* Smoke check on group queries
* Smoke check on session_id+window_id features (recording list filter + insight<>recordings link)
